opl/opl_sdl.c

```diff
@@ -154,6 +155,38 @@ static void AdvanceTime(unsigned int nsamples)
     SDL_UnlockMutex(callback_queue_mutex);
 }
 
+#if SDL_BYTEORDER == SDL_LIL_ENDIAN
+  #define OPL_SHORT SDL_SwapLE16
+#else
+  #define OPL_SHORT SDL_SwapBE16
+#endif
+
+int opl_gain = 200;
+
+static void MixAudioFormat(Uint8 *dst, const Uint8 *src, Uint32 len)
+{
+    Sint16 src1, src2;
+    int dst_sample;
+
+    while (len--)
+    {
+        src1 = OPL_SHORT(*(Sint16 *)src);
```

Comment:
Why do we read two shorts and then *add* them? Is this some kind of stereo->mono conversion? I would have expected something like `(src1 << 16) | src2`. 🤷

Reply:
The reason I am asking is because I thought about using a lookup table with amplification values, but for this the shorts would have to get normalized e.g. into the char regime, e.g. 
```c
dst_sample = (src1 + src2) * lut[(src1 + src2) >> 8] / 256;
```

Reply:
But `src2` is from `dst` output buffer. I copied this function from SDL2: https://github.com/libsdl-org/SDL/blob/39eab4bf44472790dddf02490ac260f810c30c9a/src/audio/SDL_mixer.c#L85 This code was here before.

Reply:
Ah, sorry, I am blind. 🙈  So, it does some sort of interpolation with the dest buffer.

Reply:
```
// OPL output is generated into temporary buffer and then mixed
// (to avoid overflows etc.)
```
I guess we don't need the full SDL2 function here, we can simplify it even more.

Reply:
I am asking myself what could already be in the output buffer? The sound effects?

Reply:
Oh you're probably right. I should apply it only to `src1`.